src/Routing/RouteCollection.php

```diff
@@ -418,14 +418,11 @@ public function setExtensions(array $extensions, $merge = true)
      * scope or any child scopes that share the same RouteCollection.
      *
      * @param string $name The name of the middleware. Used when applying middleware to a scope.
-     * @param callable $middleware The middleware object to register.
+     * @param callable|string $middleware The middleware callable or class name to register.
      * @return $this
      */
-    public function registerMiddleware($name, callable $middleware)
+    public function registerMiddleware($name, $middleware)
     {
-        if (is_string($middleware)) {
-            throw new RuntimeException("The '$name' middleware is not a callable object.");
-        }
```

Comment:
Should we check that the argument is a non-string callable or a valid class name and error out early?

Reply:
I don't think that's necessary.  MiddlewareQueue also accepts middleware name like `Foo` which it expands to class name using App::className() when the queue is run.

Reply:
ok